### PR TITLE
Fix memory leak when deinitializing display feeder tasks

### PR DESCRIPTION
**Pull request by vroland**
Towards #269 .

### CODE REVIEW

**Comment by vroland** [comment]
@aceisace Can you check if this fixes the issue?

**Comment by aceisace** [comment]
@vroland Yes sure, but I'll need a few days before I can test this due to some refactoring that is required when using the latest version of epdiy

**Comment by martinberlin** [comment]
Looking good.
Will test it tonight!

**Review by martinberlin — APPROVED** [review]
Approved, just wait please for @aceisace check to see if it fixes his issue

File: test/test_initialization.c [hunk 1]
```diff
@@ -35,11 +36,14 @@ TEST_CASE("re-initialization works", "[epdiy,e2e]")
...
+    int before_init = esp_get_free_internal_heap_size();
     epd_init(&TEST_BOARD, &ED097TC2, EPD_OPTIONS_DEFAULT);
  
     epd_poweron();
     vTaskDelay(2);
     epd_poweroff();
 
     epd_deinit();
+    int after_init = esp_get_free_internal_heap_size();
+    TEST_ASSERT_EQUAL(after_init, before_init);
```
Comment (vroland) [inline test/test_initialization.c:48]: @martinberlin This is what's tested here (excluding the draw)

**Comment by aceisace** [comment]
@vroland 
Hey, I finally had time to test this out today, but sadly, it does not fix the issue. This is the error this time:
```
assert failed: lq_init line_queue.c:23 (queue.bufs != NULL)

Backtrace: ... 
0x400823f9: panic_abort at ../esp/v4.4.6/esp-idf/components/esp_system/panic.c:408
0x400926a9: esp_system_abort at ../esp/v4.4.6/esp-idf/components/esp_system/esp_system.c:137
0x40099e09: __assert_func at ../esp/v4.4.6/esp-idf/components/newlib/assert.c:85
0x400f78b9: lq_init at ../esp-idf/test/components/epdiy-main/src/output_common/line_queue.c:23 (discriminator 1)
0x400f6bf1: epd_renderer_init at ../esp-idf/test/components/epdiy-main/src/render.c:322
0x400f6949: epd_init at ../esp-idf/test/components/epdiy-main/src/epdiy.c:491
0x400e74d2: loopTask(void*) at ../esp-idf/test/components/arduino/cores/esp32/main.cpp:42
0x40095d35: vPortTaskWrapper at ../esp/v4.4.6/esp-idf/components/freertos/port/xtensa/port.c:142
```

And the error is caused on this line:
```
epd_init(&epd_board_v5, &ED133UT2, EPD_LUT_64K);
```
It basically fails on the first initialisation of epdiy.

Additional Details:
- esp-idf version 4.4.6
- v5 board

**Comment by martinberlin** [comment]
That’s really strange in v7 works. Actually it is only on deinitialization. 
Are you sure nothing at all is changed in your side of the things?
Wanted to reproduce it but I gave out all my v5 boards

**Pull request merged**